### PR TITLE
Expand castle metadata and add hero metadata for the map format

### PR DESCRIPTION
**Pull request by ihhub**
relates to #6845

### CODE REVIEW

**Comment by oleg-derevenetz** [comment]
Hi @ihhub it seems that you have accidentally pushed [this commit](https://github.com/ihhub/fheroes2/commit/b1d1e7c82e5b5da4826fe4d53c84954339ddc714) to `master` instead of this PR.

**Comment by ihhub** [comment]
@oleg-derevenetz , crap. I remember that I blocked pushes to the master branch :( Let me undo things.

**Comment by ihhub** [comment]
@oleg-derevenetz , thanks a lot for pointing this out! It looks like I can still push changes to the master branch directly which I shouldn't have access to but GitHub allows it :(

**Comment by oleg-derevenetz** [comment]
@ihhub IIRC there is a separate checkbox to disable this for project owners/admins, and by default these restrictions do not apply to them.

**Comment by ihhub** [comment]
> @ihhub IIRC there is a separate checkbox to disable this for project owners/admins, and by default these restrictions do not apply to them.

I've just done it:
image

File: src/fheroes2/maps/map_format_info.cpp [hunk 1]
```diff
@@ -34,6 +35,22 @@ namespace
 
     // This value is set to avoid any corrupted files to be processed.
     const size_t minFileSize{ 128 };
+
+    template <typename TType, size_t TSize>
+    void readArray( StreamBase & stream, std::array<TType, TSize> & values )
```
[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/maps/map_format_info.cpp:40]: We can add a function inside `serialize.h` but it must come with the same logic as I created for map_format_info.cpp to make sure that the size is correct.

**Review by Districh-ru — APPROVED** [review]

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**